massa-execution-worker/src/execution.rs

```diff
@@ -248,7 +242,75 @@ impl ExecutionState {
         if slot > max_slot {
             panic!("cannot execute at a slot beyond active cursor + 1");
         }
+    }
+
+    /// Lazily query (from end to beginning) the active balance of an address at a given slot.
+    /// Returns None if the address balance could not be determined from the active history.
+    pub fn get_active_balance_at_slot(&self, slot: Slot, addr: &Address) -> Option<Amount> {
+        self.verify_active_slot(slot);
+
+        let iter = self
```

Comment:
better performance can be achieved with

* finding highest history index for which `output.slot > slot` using constant complexity because we know the 1st slot of the history and we know that the history is composed of strictly consecutive slots
* iterating from that index (non-included) backwards down to 0

Reply:
I don't understand how we can find the highest history index for which `output.slot > slot` given that `active_history` is of type `VecDeque`. I don't think it's possible to achieve constant complexity search with that data structure but maybe I'm wrong?